server/protocol.c

```diff
@@ -1490,8 +1318,59 @@ request_rec *ap_read_request(conn_rec *conn)
 
     ap_run_pre_read_request(r, conn);
 
-    /* Get the request... */
-    if (!read_request_line(r, tmp_bb) || !ap_parse_request_line(r)) {
+    r->request_time = apr_time_now();
+    rv = ap_get_brigade(r->proto_input_filters, tmp_bb, AP_MODE_READBYTES, APR_BLOCK_READ, 0);
+    if (rv != APR_SUCCESS || APR_BRIGADE_EMPTY(tmp_bb)) {
+        /* Not worth dying with. */
+        conn->keepalive = AP_CONN_CLOSE;
+        apr_pool_destroy(r->pool);
+        goto ignore;
+    }
+
+    for (e = APR_BRIGADE_FIRST(tmp_bb);
+         e != APR_BRIGADE_SENTINEL(tmp_bb);
+         e = APR_BUCKET_NEXT(e))
+    {
+        if (AP_BUCKET_IS_REQUEST(e)) {
+            if (!breq) breq = e->data;
+        }
+        else if (AP_BUCKET_IS_ERROR(e)) {
+            access_status = ((ap_bucket_error *)(e->data))->status;
+            goto die_unusable_input;
+        }
```

Comment:
The filter detected an error that should result in a HTTP response with the error status code.

Reply:
Ah OK, so the input error bucket is caught here, why not.
So up to ap_read_request() some special input filtering is in place and after that only body filtering right?
Could you please show the output of `dump_filters r->[proto_]input_filters` (from our gdbinit) at different ap_read_request() steps? This'd possibly help have a picture here, or however you'd like to describe it :)

Reply:
Added a macro in util_filter.h to dump a filter chain and now see in the logs for a http/1.1 request:

```
filter-dump before pre_read[0]: reqtimeout [58 CONNECTION]
filter-dump before pre_read[1]: log_input_output [59 CONNECTION]
filter-dump before pre_read[2]: core_in [60 NETWORK]
filter-dump after pre_read[0]: http1_request_in [30 PROTOCOL]
filter-dump after pre_read[1]: reqtimeout [58 CONNECTION]
filter-dump after pre_read[2]: log_input_output [59 CONNECTION]
filter-dump after pre_read[3]: core_in [60 NETWORK]
ap_tokenize_request_line: 'GET / HTTP/1.1'
ap_tokenize_request: error=0, method=GET, uri=/, protocol=HTTP/1.1
http1 request and headers parsed: GET / HTTP/1.1
filter-dump after get_brigade[0]: reqtimeout [58 CONNECTION]
filter-dump after get_brigade[1]: log_input_output [59 CONNECTION]
filter-dump after get_brigade[2]: core_in [60 NETWORK]
checking request: GET / HTTP/1.1
assigned protocol: HTTP/1.1, error=0
assigned request_line: GET / HTTP/1.1, error=0
parsed uri: r->status=200, error=0
filter-dump after post_read[0]: http_in [30 PROTOCOL]
filter-dump after post_read[1]: http1_body_in [40 TRANSCODE]
filter-dump after post_read[2]: reqtimeout [58 CONNECTION]
filter-dump after post_read[3]: log_input_output [59 CONNECTION]
filter-dump after post_read[4]: core_in [60 NETWORK]
```